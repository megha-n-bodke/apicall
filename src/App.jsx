import { useEffect, useState } from "react";
import reactLogo from "./assets/react.svg";
import "./App.css";
import axios from "axios";

function App() {
  const [res, setRes] = useState([]);
  useEffect(() => {
    displayData();
  }, []);
  async function displayData() {
    const baseUrl = "https://api.github.com/users/megha-n-bodke";
    const { data } = await axios.get(baseUrl);
    setRes(data);
  }
  console.log(res);
  return (
    <div className="App">
      <h1>My Github Profile</h1>
      <>
        <h2>{res.name}</h2>
        <img src={res.avatar_url} alt="Profile" />
        <p>{res.bio}</p>
        <small>Public Repositories:{res.public_repos}</small>
      </>
    </div>
  );
}

export default App;
//abhishek@incubyte.co
