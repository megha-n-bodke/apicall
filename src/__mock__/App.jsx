import React from "react";

const App = () => {
  return (
    <div className="App">
      <h1>My Github Profile</h1>
      <>
        <h2>Megha Bodke</h2>
        <img
          src="https://avatars.githubusercontent.com/u/13938070?v=4"
          alt="Profile"
        />
        <p>
          Software engineer | Full-stack developer | Stack:HTML | CSS |
          JavaScript | React | Redux | Ruby | Ruby on Rails. Open For New Job
          Opportunities.
        </p>
        <small>Public Repositories:62</small>
      </>
    </div>
  );
};

export default App;
