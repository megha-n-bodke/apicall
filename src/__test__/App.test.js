import App from "../__mock__/App";
import renderer from "react-test-renderer";

describe("App", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
