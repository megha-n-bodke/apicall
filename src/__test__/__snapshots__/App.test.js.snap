// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    My Github Profile
  </h1>
  <h2>
    Megha Bodke
  </h2>
  <img
    alt="Profile"
    src="https://avatars.githubusercontent.com/u/13938070?v=4"
  />
  <p>
    Software engineer | Full-stack developer | Stack:HTML | CSS | JavaScript | React | Redux | Ruby | Ruby on Rails. Open For New Job Opportunities.
  </p>
  <small>
    Public Repositories:62
  </small>
</div>
`;
